internal/services/netapp/netapp_backup_policy_test.go

```diff
@@ -0,0 +1,163 @@
+				check.That(data.ResourceName).Key("daily_backups_to_keep").HasValue("2"),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccNetAppBackupPolicy_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_netapp_backup_policy", "test")
+	r := NetAppBackupPolicyResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("tags.%").HasValue("1"),
+				check.That(data.ResourceName).Key("daily_backups_to_keep").HasValue("2"),
+				check.That(data.ResourceName).Key("daily_backups_to_keep").HasValue("2"),
+				check.That(data.ResourceName).Key("daily_backups_to_keep").HasValue("2"),
+				check.That(data.ResourceName).Key("enabled").HasValue("true"),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.update(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("tags.%").HasValue("2"),
+				check.That(data.ResourceName).Key("daily_backups_to_keep").HasValue("10"),
+				check.That(data.ResourceName).Key("daily_backups_to_keep").HasValue("10"),
+				check.That(data.ResourceName).Key("daily_backups_to_keep").HasValue("10"),
+				check.That(data.ResourceName).Key("enabled").HasValue("false"),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (r NetAppBackupPolicyResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%[1]s
+
+resource "azurerm_netapp_backup_policy" "test" {
+  name                    = "acctest-NetAppBackupPolicy-%[2]d"
+  resource_group_name     = azurerm_resource_group.test.name
+  location                = azurerm_resource_group.test.location
+  account_name            = azurerm_netapp_account.test.name
+  daily_backups_to_keep   = 2
+  weekly_backups_to_keep  = 2
+  monthly_backups_to_keep = 2
+  enabled                 = true
+
+  tags = {
+    "testTag" = "testTagValue"
+  }
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r NetAppBackupPolicyResource) update(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%[1]s
+
+resource "azurerm_netapp_backup_policy" "test" {
+  name                    = "acctest-NetAppBackupPolicy-%[2]d"
+  resource_group_name     = azurerm_resource_group.test.name
+  location                = azurerm_resource_group.test.location
+  account_name            = azurerm_netapp_account.test.name
+  daily_backups_to_keep   = 10
+  weekly_backups_to_keep  = 10
+  monthly_backups_to_keep = 10
+  enabled                 = false
+
+  tags = {
+    "testTag" = "testTagValue",
+    "FoO"     = "BaR"
+  }
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (NetAppBackupPolicyResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {
+    resource_group {
+      prevent_deletion_if_contains_resources = false
+    }
+  }
```

Comment:
This shouldn't be set in acceptance tests, if it's required for the test to pass then that usually indicates that there is an issue in the API around deletion that we should look into.

```suggestion
  features {}
```

Reply:
This should stay, it is for tests to pass given Microsoft Azure team forces a few Azure Policies on service team's subscriptions and automatic resources get's created on resource groups depending on what resource types we're deploying.

If I remove this, tests will fail and this will cause waste of time on my side and I can't have that.